File: daily-exercise/special_topic/dp/longest-continuous-increasing-subsequence.go
func findLengthOfLCIS(nums []int) int {
    if len(nums) <= 1{
        return len(nums)
    }
    dp := make([]int,len(nums))
    for i := 0; i < len(dp); i++ {
		dp[i] = 1
	}
    res := 1
    for i:=1;i<len(nums);i++{
        if nums[i] > nums[i-1]{
            dp[i] = dp[i-1] + 1
        }
        res = max(res, dp[i])
    }
    return res
}

func max(a,b int) int {
    if a > b{
        return a
    }
    return b
}

File: daily-exercise/special_topic/dp/longest-continuous-increasing-subsequence.py
# -*- coding: utf-8 -*-
# @Time    : 2022/8/8 20:33
# @Author  : kevin
# @Version : python 3.7
# @Desc    : 最长连续递增子序列

# 概括来说：不连续递增子序列的跟前0-i 个状态有关，连续递增的子序列只跟前一个状态有关

class Solution:
    def findLengthOfLCIS(self, nums):
        if len(nums) <= 1:
            return len(nums)
        dp = [1] * len(nums)
        res = 1
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                dp[i] = dp[i - 1] + 1
            res = max(res, dp[i])
        return res
